tools/azure-sdk-tools/ci_tools/conda/CondaConfiguration.py

```diff
@@ -0,0 +1,196 @@
+        raise e
+    filename = os.path.basename(target_zip)
+
+    if filename.endswith("tar.gz") or filename.endswith(".zip"):
+        return target_zip
+    else:
+        raise ValueError(f"Unable to find a source distribution for {package}@{version}.")
+
+
+class CheckoutConfiguration:
+    def __init__(self, raw_json: dict):
+        # we should always have a package name
+
+        if "package" in raw_json:
+            self.package = raw_json["package"]
+        else:
+            raise ValueError("A checkout configuration MUST have a package name defined in key 'package'.")
+
+        if "checkout_path" in raw_json:
+            self.checkout_path = raw_json["checkout_path"]
+        else:
+            self.checkout_path = None
+
+        if "version" in raw_json:
+            self.version = raw_json["version"]
+        else:
+            self.version = None
+
+        if "download_uri" in raw_json:
+            self.download_uri = raw_json["download_uri"]
+        else:
+            self.download_uri = None
+
+        if "version" in raw_json and self.checkout_path is None:
+            self.download_uri = get_package_sdist_url(self.package, self.version)
+
+        if not self.checkout_path and not self.download_uri:
+            raise ValueError(
+                "When defining a checkout configuration, one must either have a valid PyPI download url"
+                " (download_uri) or a path and version in the repo (checkout_path, version)."
+            )
+
+    def __str__(self) -> str:
+        if self.download_uri:
+            return f"""- {self.package} downloaded from pypi
+  {self.download_uri}"""
+        else:
+            return f"""- {self.checkout_path}/{self.package} from git @ {self.version}"""
+
+
+def parse_checkout_config(checkout_configs: List[Any]) -> List[CheckoutConfiguration]:
+    configs = []
+    for checkout_config in checkout_configs:
+        configs.append(CheckoutConfiguration(checkout_config))
+
+    return configs
+
+
+class CondaConfiguration:
+    def __init__(
+        self,
+        name: str,
+        common_root: str,
+        in_batch: bool,
+        checkout: List[CheckoutConfiguration],
+        created_sdist_path: str = None,
+        service: str = "",
+        conda_py_versions: List[str] = [],
+        channels: List[str] = []
+    ):
+        self.name: str = name
+        self.common_root: str = common_root
+        self.in_batch: bool = in_batch
+        self.checkout: List[CheckoutConfiguration] = checkout
+        self.created_sdist_path: str = created_sdist_path
+        self.service: str = service
+        self.conda_py_versions = conda_py_versions
+        self.channels = channels
+
+    @classmethod
+    def from_json(cls, raw_json_blob: dict):
+        name = raw_json_blob["name"]
+        common_root = None
+        service = None
+
+        if "common_root" in raw_json_blob:
+            common_root = raw_json_blob["common_root"]
```

Comment:
Another opportunity for `.get()`

Reply:
I like...blanked on that pattern entirely using this...for some reason. I'll fix all of these.